std/traits.d

```diff
@@ -6730,3 +6730,29 @@ unittest
     static assert(getSymbolsByUDA!(C, UDA)[0].stringof == "C");
     static assert(getSymbolsByUDA!(C, UDA)[1].stringof == "d");
 }
+
+/**
+   Returns: $(D true) iff all types $(D T) are the same.
+*/
+template allSameType(T...)
+{
+    static if (T.length <= 1)
+    {
+        enum bool allSameType = true;
+    }
+    else
+    {
+        enum bool allSameType = is(T[0] == T[1]) && allSameType!(T[1..$]);
+    }
+}
+
```

Comment:
Variant

```D
enum allSameType(T...) = !T.length || (is(T[0] == T[T.length > 1]) && allSameType!(T[1 .. $]));
```

fails to compile 

```D
traits_ex.d(97,71): Error: slice [1..0] is out of range of [0..0]
traits_ex.d(97,71): Error: template instance traits_ex.allSameType!() error instantiating
traits_ex.d(97,71):        instantiated from here: allSameType!int
traits_ex.d(109,19):        instantiated from here: allSameType!(int, int)
traits_ex.d(97,71): Error: template instance traits_ex.allSameType!double error instantiating
traits_ex.d(110,20):        instantiated from here: allSameType!(int, double)
traits_ex.d(111,20): Error: template instance traits_ex.allSameType!(int, int, double) error instantiating
traits_ex.d(183,5): Error: static assert  (is(typeof(__error) == typeof(arr))) is false
```

I stick with my current version for now.